File: src/webchat/app/json_models/Message.java
package json_models;

import java.util.ArrayList;
import java.util.List;

import org.codehaus.jackson.JsonNode;

import play.libs.Json;

import models.Channel;
import flexjson.JSONException;
import flexjson.JSONSerializer;


public class Message{
	public final String type;
	public MessageData data;
	
	public Message(){
		this.type = "message";
	}
	
	public static JsonNode genMessage(JsonNode inmessage)
	{
		String json = "";
		try{
			
			Message m = new Message();
			List<Channel> channels = new ArrayList<Channel>();
			m.data.message = inmessage.findPath("message").asText();
			//m.data.channels = channels;
			
			JSONSerializer mser = new JSONSerializer();
			json = mser.exclude(".class").serialize(m);
		}catch (JSONException e){
			e.printStackTrace();
		}
		return Json.parse(json);
	}
}

File: src/webchat/public/javascripts/toaster/messages/message.js
(function() {
  var webchat;

  webchat = require('webchat');

  webchat.Message = (function() {

    function _Class(type) {
      this.type = type;
    }

    _Class.prototype.serialize = function(data) {
      var message;
      message = {
        type: this.type,
        data: data
      };
      return message;
    };

    return _Class;

  })();

}).call(this);
